### PR TITLE
Adding comments to TestReadPublicKey

### PR DESCRIPTION
**Pull request by matt-tsai**
Writing contribution document and am commenting on this test as an example.

### CODE REVIEW

File: direct/tpm2/read_public_test.go [hunk 1]
```diff
@@ -51,26 +57,41 @@ func TestReadPublicKey(t *testing.T) {
...
+	// Executing the command uses reflection to pack the bytes into a 
+	// TPM2_ReadPublic command, returns a TPM2_ReadPublic Response.
+	// This response is also decoded so you are again working with structs 
+	// that can be found in Part 3, Commands, section 12.4.
 	rspRP, err := readPublic.Execute(thetpm)
 	if err != nil {
 		t.Fatalf("ReadPublic failed: %v", err)
 	}
 
+	// Compare the Unique portion of the PublicAreas to ensure they are equal.
```
Comment (alexmwu) [inline direct/tpm2/read_public_test.go:90]: Might be useful to explain that the Unique represents the public part of the key.

File: direct/tpm2/read_public_test.go [hunk 2]
```diff
@@ -51,26 +57,41 @@ func TestReadPublicKey(t *testing.T) {
...
+	// Notice how this test uses off-tpm verification of hardcoded a PublicArea
```
Comment (alexmwu) [inline direct/tpm2/read_public_test.go:91]: remove this line.

File: direct/tpm2/read_public_test.go [hunk 3]
```diff
@@ -13,14 +13,20 @@ import (
...
+// the PublicArea read from executing readPublic.
 func TestReadPublicKey(t *testing.T) {
-
+	// Open simulated TPM for testing.
 	thetpm, err := simulator.OpenSimulator()
 	if err != nil {
 		t.Fatalf("could not connect to TPM simulator: %v", err)
 	}
+	
+	// Defer the close of the simulated TPM to after use.
```
Comment (alexmwu) [inline direct/tpm2/read_public_test.go:25]: Without this, other programs/tests may not be able to get a handle to the TPM.

File: direct/tpm2/read_public_test.go [hunk 4]
```diff
@@ -51,26 +57,41 @@ func TestReadPublicKey(t *testing.T) {
...
+	// Executing the command uses reflection to pack the bytes into a 
+	// TPM2_CreatePrimary command, returns a TPM2_CreatePrimary Response.
+	// This response is also decoded so you are again working with structs 
+	// that can be found in Part 3, Commands, section 24.1.
 	rspCP, err := createPrimary.Execute(thetpm)
 	if err != nil {
 		t.Fatalf("CreatePrimary failed: %v", err)
 	}
 
+	// The TPM can only hold so much in nonvolitile memory, thus we must
```
Comment (alexmwu) [inline direct/tpm2/read_public_test.go:69]: nit: nonvolatile

File: direct/tpm2/read_public_test.go [hunk 5]
```diff
@@ -51,26 +57,41 @@ func TestReadPublicKey(t *testing.T) {
...
+	// flush the handle after we are done using it to prevent overloading. 
+	// Again we defer the flush to after we are done using the object.
```
Comment (alexmwu) [inline direct/tpm2/read_public_test.go:71]: It is generally good practice to `defer` the cleanup immediately after loading an object or creating an Authorization Session.

File: direct/tpm2/read_public_test.go [hunk 6]
```diff
@@ -13,14 +13,20 @@ import (
...
 	if err != nil {
 		t.Fatalf("could not connect to TPM simulator: %v", err)
 	}
+	
+	// Defer the close of the simulated TPM to after use.
 	defer thetpm.Close()
 
+	// Fill in the CreatePrimary struct.
+	// See definition in Part 3, Commands, section 24.1.
 	createPrimary := CreatePrimary{
```
Comment (alexmwu) [inline direct/tpm2/read_public_test.go:30]: did we ever create test helper templates or Publics? If so, add a pointer to them.

**Review by alexmwu — CHANGES_REQUESTED** [review]

File: direct/tpm2/read_public_test.go (hunk 1 above)
Comment (matt-tsai) [reply direct/tpm2/read_public_test.go:90]: Resolved.

File: direct/tpm2/read_public_test.go (hunk 2 above)
Comment (matt-tsai) [reply direct/tpm2/read_public_test.go:91]: Resolved.

File: direct/tpm2/read_public_test.go (hunk 3 above)
Comment (matt-tsai) [reply direct/tpm2/read_public_test.go:25]: Comment added.

File: direct/tpm2/read_public_test.go (hunk 4 above)
Comment (matt-tsai) [reply direct/tpm2/read_public_test.go:69]: Good catch, resolved.

File: direct/tpm2/read_public_test.go (hunk 5 above)
Comment (matt-tsai) [reply direct/tpm2/read_public_test.go:71]: Nice comment, also added more context from Part 1, resolved.

File: direct/tpm2/read_public_test.go (hunk 6 above)
Comment (matt-tsai) [reply direct/tpm2/read_public_test.go:30]: Added. Resolved.

File: direct/tpm2/read_public_test.go [hunk 7]
```diff
@@ -13,14 +13,20 @@ import (
 	"github.com/google/go-tpm/direct/transport/simulator"
 )
 
+// TestReadPublicKey compares the createPrimary PublicArea when instantiated with
```
[the remark this answers is not in the document]
Comment (matt-tsai) [reply direct/tpm2/read_public_test.go:16]: Resolved.

**Review by alexmwu — APPROVED** [review]

**Pull request merged**